components/applications-service/pkg/storage/postgres/db.go

```diff
@@ -111,13 +107,6 @@ type service struct {
 	HealthUpdatedAt     time.Time `db:"health_updated_at"`
 	CreatedAt           time.Time `db:"-"`
 	UpdatedAt           time.Time `db:"-"`
-
-	// (internal) use it to know if the service needs an update or not
-	needUpdate bool `db:"-"`
-}
-
-func (svc *service) NeedUpdate() bool {
-	return svc.needUpdate
```

Comment:
We update the service every time so we can find when it disconnects, so this was just always returning true in practice.